bluepyopt/ephys/stimuli.py

```diff
@@ -140,7 +141,7 @@ def instantiate(self, sim=None, icell=None):
                 netcon = sim.neuron.h.NetCon(netstim, synapse)
                 netcon.weight[0] = self.weight
 
-                self.connections[location.name] = (netcon, netstim)
+                self.connections[location.name].append((netcon, netstim))
```

Comment:
Ok, that makes sense. Didn't think about the use case in which the same location has multiple netstims.

Reply:
Well actually it's a bit weird: I have a NrnModPointProcessMechanism object that represents a synapse model at multiple locations (since it has a list of NrnSeclistCompLocation). Then when you make a NrnPointProcessLocation object for that point process it covers multiple synapses and their location. So when you instantiate it, you want the NetCons to all those synapses to survive.

Before this change, my NetCons tal all except the last synapse would disappear so they wouldn't receive events. Even if they all have the same NetStim.

Reply:
What I mean by weird is that it's a bit counterintuitive: one NrnPointProcessLocation can actually cover multiple synapses at distinct locations.

I know you could make a separate NrnMODPointProcessMechanism for each individual synapse, each with their own NrnPointProcessLocation, but then it's harder to give them all the same parameters.

Reply:
Yeah, sure, makes sense